internal/core/propfilters/dimpropfilter.go

```diff
@@ -90,15 +90,11 @@ func (f *basicDimPropsFilter) FilterDimProps(dimProps *types.DimProperties) *typ
 		filteredProperties = dimProps.Properties
 	}
 
-	if len(filteredProperties) > 0 {
-		return &types.DimProperties{
-			Dimension:  dimProps.Dimension,
-			Properties: filteredProperties,
-			Tags:       dimProps.Tags,
-		}
+	return &types.DimProperties{
+		Dimension:  dimProps.Dimension,
+		Properties: filteredProperties,
+		Tags:       dimProps.Tags,
 	}
-
-	return nil
```

Comment:
@mstumpfx instead of it "nilling out" the dimension set and becoming a noop in the writer, I'm making it just return a dimension with an empty property list to handle the case where a dimension has one set of properties (not filtered) and then changes to another set of properties that are all filtered, so we still do the update to the backend.  I think there was also a bug before where if all the properties were filtered out, no tags would sync since we don't consider the number of tags.  Let me know if you can think of any other edge cases with this that I might not be handling.

Reply:
I guess we should consider a `tagsToExclude` config option if some of our monitors are going to start emitted tags by default.

Reply:
yeah I don't really want to go there yet.